upload/extension/opencart/catalog/model/payment/opayo.php

```diff
@@ -238,40 +238,28 @@ private function getOrderTransactions(int $opayo_order_id): array {
 	/**
 	 * Subscription Payment
 	 *
-	 * @param array<string, mixed>  $item
-	 * @param string $vendor_tx_code
+	 * @param array<string, mixed> $item
+	 * @param string               $vendor_tx_code
 	 *
 	 * @return void
 	 */
 	public function subscriptionPayment(array $item, string $vendor_tx_code): void {
 		$this->load->model('checkout/subscription');
+		$this->load->model('checkout/order');
 		$this->load->model('extension/payment/opayo');
 
 		if ($item['subscription']['trial'] == 1) {
 			$price = $item['subscription']['trial_price'];
-			$trial_amt = $this->currency->format($this->tax->calculate($item['subscription']['trial_price'], $item['tax_class_id'], $this->config->get('config_tax')), $this->session->data['currency'], false, false) * $item['quantity'] . ' ' . $this->session->data['currency'];
-			$trial_text = sprintf($this->language->get('text_trial'), $trial_amt, $item['subscription']['trial_cycle'], $item['subscription']['trial_frequency'], $item['subscription']['trial_duration']);
 		} else {
 			$price = $item['subscription']['price'];
-			$trial_text = '';
-		}
-
-		$subscription_amt = $this->currency->format($this->tax->calculate($item['subscription']['price'], $item['tax_class_id'], $this->config->get('config_tax')), $this->session->data['currency'], false, false) * $item['quantity'] . ' ' . $this->session->data['currency'];
-		$subscription_description = $trial_text . sprintf($this->language->get('text_subscription'), $subscription_amt, $item['subscription']['cycle'], $item['subscription']['frequency']);
-
-		if ($item['subscription']['duration'] > 0) {
-			$subscription_description .= sprintf($this->language->get('text_length'), $item['subscription']['duration']);
 		}
 
-		$item['subscription']['description'] = $subscription_description;
+		$order_info = $this->model_checkout_order->getOrder($this->session->data['order_id']);
 
-		// Create new subscription and set to pending status as no payment has been made yet.
-		$subscription_id = $this->model_checkout_subscription->addSubscription($this->session->data['order_id'], $item['subscription']);
+		$subscription_id = $order_info['subscription_id'];
 
 		$this->model_checkout_subscription->editReference($subscription_id, $vendor_tx_code);
```

Comment:
@TheCartpenter there is no such function, can you try and figure out what this should be replaced with?

Reply:
Seem to be a leftover from the original developer. I looked everyone for this method but the reference field no longer exists in the core nor do any extensions on the opencart-3 except ```securetrading_pp``` and the old recurring system. I'll remove that line.